### PR TITLE
Upgrade to zookeeper 3.6.1

### PR DESCRIPTION
**Pull request by amuraru**
Fixes: https://github.com/pravega/zookeeper-operator/issues/128, https://github.com/pravega/zookeeper-operator/issues/155, https://github.com/pravega/zookeeper-operator/issues/164

1. Upgrades `zu` utility to compile againstzookeeper 3.6.1

2. Upgrades the base image for pravega/zookeeper to zookeeper:3.6.1

3. Enable prometheus monitoring for zookeeper pods

Since 3.6.0 ZooKeeper supports native Prometheus monitoring
https://github.com/apache/zookeeper/blob/master/zookeeper-docs/src/main/resources/markdown/zookeeperMonitor.md#prometheus

This patch
- enables `PrometheusMetricsProvider` in zoo.cfg
- expose a new `metrics` port to ZK pods, ZK services
- disambiguate the headless and client service labels by adding a new label,
 so it can be used in Prometheus `ServiceMonitors`

For users using Prometheus Operator for monitoring the following
ServiceMonitor CR can be created to enable ZK cluster monitoring
deployed in `zookeeper` namespace

```yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: zookeeper-servicemonitor
  labels:
    app: zookeeper
    release: monitoring
spec:
  selector:
    matchLabels:
      app: zk
  namespaceSelector:
    matchNames:
    - zookeeper
  endpoints:
  - interval: 10s
    port: metrics
    path: /metrics
    honorLabels: true
    relabelings:
    - sourceLabels: [__meta_kubernetes_pod_ip]
      separator: ;
      regex: (.*)
      targetLabel: pod_ip
      replacement: $1
      action: replace
    - sourceLabels: [__meta_kubernetes_pod_name]
      separator: ;
      regex: (.*)
      targetLabel: pod_name
      replacement: $1
      action: replace
```

### CODE REVIEW

**Comment by marvinmarnold** [comment]
+1 looking forward to this getting merged

**Comment by amuraru** [comment]
rebased on top of master - to trigger new e2e tests

File: docker/zu/gradlew [hunk 1]
```diff
@@ -159,14 +175,9 @@ save () {
     for i do printf %s\\n "$i" | sed "s/'/'\\\\''/g;1s/^/'/;\$s/\$/' \\\\/" ; done
     echo " "
 }
-APP_ARGS=$(save "$@")
+APP_ARGS=`save "$@"`
 
 # Collect all arguments for the java command, following the shell quoting and substitution rules
 eval set -- $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS "\"-Dorg.gradle.appname=$APP_BASE_NAME\"" -classpath "\"$CLASSPATH\"" org.gradle.wrapper.GradleWrapperMain "$APP_ARGS"
 
-# by default we should be in the correct project dir, but when run from Finder on Mac, the cwd is wrong
```
Comment (anishakj) [inline docker/zu/gradlew:167]: why are we removing these?

Comment (amuraru) [reply docker/zu/gradlew:167]: this is auto-generated by `gradle wrapper` and came as part of upgrading gradle from 4.8.1 to 6.3 in zu sub-project
See: https://github.com/pravega/zookeeper-operator/pull/175/files#diff-efc13f2a48bac60893fac008f48da0f7R3

File: docker/zu/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
```
[the remark this answers is not in the document]
Comment (amuraru) [reply docker/zu/build.gradle.kts:1]: zu is already kotlin so thought kotlin gradle dsl makes more sense.
The build script is much simpler if you check this: https://github.com/pravega/zookeeper-operator/blob/dede2bacf9ff56fa85bd81286cea0fd20815e632/docker/zu/build.gradle.kts

**Pull request merged**